### PR TITLE
Scaling policies optimizer

### PR DESCRIPTION
**Pull request by perezp**
With this PR the optimizer reads the `autoscale` value from AAM even if the AAM does not contain  `QoSinfo` . Tests for this case have been added. 

Before this PR, when it was found the lack of the required quality information `QoSinfo` the `autoscale` value was not read but it took the default value of `true`

### CODE REVIEW

File: planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/Topology.java [hunk 1]
```diff
@@ -26,10 +26,10 @@
 
 public class Topology {
 
-   static Logger log = LoggerFactory.getLogger(Topology.class);
+   static Logger                 log = LoggerFactory.getLogger(Topology.class);
```
Comment (kiuby88) [inline planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/Topology.java]: Probably these spaces could be removed

File: planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/Topology.java [hunk 2]
```diff
@@ -26,10 +26,10 @@
...
 
    private List<TopologyElement> modules;
-   private TopologyElement initialElement;
+   private TopologyElement       initialElement;
```
Comment (kiuby88) [inline planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/Topology.java]: same above

File: planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLmodulesOptimizerParser.java [hunk 3]
```diff
@@ -357,24 +356,42 @@ private static QualityInformation parseQoS(Map<String, Object> qosInformation) {
...
-         boolean moduleCanScale = (boolean) moduleProperties.get(TOSCAkeywords.MODULE_AUTOSCALE_PROPERTY);
-         return moduleCanScale;
-      } catch (Exception e) {
+         Map<String, Object> moduleInfo;
+         if (modules.containsKey(elementName)) {
+            moduleInfo = (Map<String, Object>) modules.get(elementName);
+         } else {
+            // not specified the scalability. Default value for not specified is
+            // true;
+            return true;
```
Comment (kiuby88) [inline planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLmodulesOptimizerParser.java]: Probably it should be better that a module will not be scaling by default. WDYT?

File: planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLmodulesOptimizerParser.java [hunk 4]
```diff
@@ -357,24 +356,42 @@ private static QualityInformation parseQoS(Map<String, Object> qosInformation) {
...
+         }
+         if(moduleProperties.containsKey(TOSCAkeywords.MODULE_AUTOSCALE_PROPERTY)){
+            return (boolean) moduleProperties.get(TOSCAkeywords.MODULE_AUTOSCALE_PROPERTY);
+         }else {
+            // not specified the scalability. Default value for not specified is
+            // true;
+            return true;
```
Comment (kiuby88) [inline planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLmodulesOptimizerParser.java]: same above

File: planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLmodulesOptimizerParser.java [hunk 5]
```diff
@@ -357,24 +356,42 @@ private static QualityInformation parseQoS(Map<String, Object> qosInformation) {
...
-         if (log.isDebugEnabled()) {
-            log.debug("For module " + elementName
-                  + " it was not found information in the AAM regarding its possibility to scale");
-         }
+         log.debug("For module {} it was not found information in the AAM regarding its possibility to scale",
+               elementName);
+
          // Returning the default value "true" for seeing policies generated
       }
       return true;
```
Comment (kiuby88) [inline planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLmodulesOptimizerParser.java]: same above

**Comment by kiuby88** [comment]
Good work @perezp , just a pair of comments

File: planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLmodulesOptimizerParser.java (hunk 3 above)
Comment (perezp) [inline planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLmodulesOptimizerParser.java]: @kiuby88 Thank you for the review. 
I've addressed all your comments and pushed. 
Now the default option to scale a module is false.

**Comment by kiuby88** [comment]
+1
to merge!!!!!!!!!!!!!

**Pull request merged**